test/spectests/spectests.cpp

```diff
@@ -0,0 +1,294 @@
+#include "execute.hpp"
+#include "parser.hpp"
+#include <nlohmann/json.hpp>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+namespace fs = std::filesystem;
+using json = nlohmann::json;
+
+namespace
+{
+constexpr auto JsonExtension = ".json";
+
+template <typename T>
+uint64_t json_to_value(const json& v)
+{
+    return static_cast<uint64_t>(static_cast<T>(std::stoull(v.get<std::string>())));
+}
+
+fizzy::bytes load_wasm_file(const fs::path& path, std::string_view filename)
```

Comment:
Parameters names are confusing. What is `path`?

Reply:
renamed